aptos-move/move-examples/token_v2/sources/coin.move

```diff
@@ -0,0 +1,427 @@
+module token_v2::coin {
+    use aptos_std::smart_table::{Self, SmartTable};
+    use aptos_framework::object::{Self, Object, object_address, is_owner, generate_transfer_ref, DeleteRef, generate_delete_ref};
+    use std::signer;
+    use token_v2::common::{increase_supply, FreezeCap, assert_freeze_cap_and_asset_match, MintCap, BurnCap, decrease_supply, get_freeze_from_asset_owner_caps, put_freeze_to_asset_owner_caps, get_burn_from_asset_owner_caps, put_burn_to_asset_owner_caps, get_mint_from_asset_owner_caps, put_mint_to_asset_owner_caps};
+    use std::vector;
+    use std::error;
+    use std::signer::address_of;
+    #[test_only]
+    use token_v2::common::{create_and_initialize_test_token, get_current_supply, remove_fungible_asset_metadata, asset_owner_caps_contain_mint, asset_owner_caps_contain_freeze, asset_owner_caps_contain_burn};
+    use std::option::Option;
+    use std::option;
+
+    /// The coin resource existence error.
+    const ECOIN: u64 = 1;
+    /// Amount cannot be zero.
+    const EAMOUNT_CANNOT_BE_ZERO: u64 = 2;
+    /// Not the owner.
+    const ENOT_OWNER: u64 = 3;
+    /// CoinStore errors.
+    const ECOIN_STORE: u64 = 4;
+    /// The token account has asset mismatch.
+    const EASSET_ADDRESS_MISMATCH: u64 = 5;
+    /// The token account has positive balance so cannot be deleted.
+    const EBALANCE_NOT_ZERO: u64 = 6;
+    /// The token account is still frozen so cannot be deleted.
+    const ESTILL_FROZEN: u64 = 7;
+    /// The coin object existence error.
+    const ECOIN_OBJECT: u64 = 8;
+    /// Insufficient balance.
+    const EINSUFFICIENT_BALANCE: u64 = 9;
+    /// Coin type mismatch.
+    const ECOIN_TYPE_MISMATCH: u64 = 9;
+
+    struct CoinStore has key {
+        index: SmartTable<address, Object<Coin>>
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    struct Coin has key {
+        asset_addr: address,
```

Comment:
Can this be Object<T>?

Reply:
Then we need one CoinStore for each T...